eng/build.sh

```diff
@@ -211,33 +205,29 @@ function BuildSolution {
     quiet_restore=true
   fi
 
-  # Node reuse fails because multiple different versions of FSharp.Build.dll get loaded into MSBuild nodes
-  node_reuse=false
```

Comment:
Why is this being removed?  Node reuse was causing failures on Mac and Linux du to incompatible FSHarp.Build.dll being loaded into build nodes

Reply:
nodereuse=false is set explicitly: 
https://github.com/dotnet/fsharp/pull/6979/files#diff-75d32cbca8eddea951ce0484aa996d8fR214
https://github.com/dotnet/fsharp/pull/6979/files#diff-75d32cbca8eddea951ce0484aa996d8fR226
https://github.com/dotnet/fsharp/pull/6979/files#diff-75d32cbca8eddea951ce0484aa996d8fR247

I didn't see the point of parameterising it, incompatible fsharp.build isn't the issue I saw, however, there were locked files during the build caused by previous builds.